server/methods/browseChannels.js

```diff
@@ -132,6 +132,14 @@ const getTeams = (user, searchTerm, sort, pagination) => {
 		roomsCount: getChannelsCountForTeam(room.teamId),
 	}));
 
+	const sortTeamsByRoomCount = (a, b) => {
+		if (a.featured && !b.featured) { return -1; }
+		if (b.featured && !a.featured) { return 1; }
+		return sort.channelsCount === -1 ? b.roomsCount - a.roomsCount : a.roomsCount - b.roomsCount;
+	};
```

Comment:
Hello @Darshilp326, Does this sort would consider all the teams from the database?
When I checked this, It only sorts channels on the current page and considering only a part of the database result (which is 25 teams in my case), and also it is not displaying the team having the highest channel count at the top. 
As @KevLehman suggested, you may need to apply sorting directly on the database to fix this.

Reply:
We are not storing roomsCount in room schema. It is being queried from another place. Yes you are correct it would sort the first 25 rows only!

Reply:
What @sumukhah said was my original concern 😬 . Basically, this is not sorting the teams based on the rooms count, this is sorting just the elements from the current page, which is not kind of correct. For example, a team with 99 rooms that appears in the second page won't show it as the first item, because the sort is just considering the elements from that page.

The solution to do the sort on the database would be to use a weird aggregation pipeline (to make it work in the current state) or to add the rooms count to the team 🤔